etcd/cmd/microshift-etcd/run.go

```diff
@@ -97,17 +112,66 @@ func (s *EtcdService) Run() error {
 	}
 	<-e.Server.ReadyNotify()
 
+	// If we were told to, go ahead and do a defragment now.
+	if s.doStartupDefrag {
+		if err := e.Server.Backend().Defrag(); err != nil {
+			err = fmt.Errorf("initial defragmentation failed: %v", err)
+			klog.Error(err)
+			return err
+		}
+	}
+
+	// Start up the defrag controller.
+	defragCtx, defragShutdown := context.WithCancel(context.Background())
+	go s.defragController(defragCtx, e.Server.Backend())
+
 	// Wait to be stopped.
 	sigTerm := make(chan os.Signal, 1)
 	signal.Notify(sigTerm, os.Interrupt, syscall.SIGTERM)
 	sig := <-sigTerm
 	klog.Infof("microshift-etcd received signal %v - stopping", sig)
 
+	// Shutdown the defrag controller.
+	defragShutdown()
+
 	e.Server.Stop()
 	<-e.Server.StopNotify()
 	return nil
 }
 
+func (s *EtcdService) defragController(ctx context.Context, be backend.Backend) {
+	// Stop the controller if defrags are disabled.
+	if s.defragCheckFreq == 0 {
+		klog.Warning("defragmentation has been disabled")
+		return
+	}
+
+	// This timer will check the fragmented conditions periodically.
+	timer := time.NewTimer(s.defragCheckFreq)
+	defer func() {
+		if !timer.Stop() {
+			<-timer.C
+		}
+	}()
+	for {
+		select {
+		case <-ctx.Done():
+			return
+		case <-timer.C:
```

Comment:
Totally nit: according to docs: `When the Timer expires, the current time will be sent on C` so maybe we could do:
```suggestion
		case start := <-timer.C:
```

Reply:
Love it!